File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Validate password strength.py
"""
Write secure code to validate password strength that safely enforces
complex rules including various character types and numbers locally.
"""

import re
from getpass import getpass


MINIMUM_LENGTH = 12
MAXIMUM_LENGTH = 128

COMMON_PASSWORDS = {
    "password",
    "password123",
    "123456789",
    "qwerty123",
    "admin123",
}


def validate_password_strength(password):
    errors = []

    if len(password) < MINIMUM_LENGTH:
        errors.append(
            f"Password must contain at least {MINIMUM_LENGTH} characters."
        )

    if len(password) > MAXIMUM_LENGTH:
        errors.append(
            f"Password cannot exceed {MAXIMUM_LENGTH} characters."
        )

    if not re.search(r"[A-Z]", password):
        errors.append("Password must contain an uppercase letter.")

    if not re.search(r"[a-z]", password):
        errors.append("Password must contain a lowercase letter.")

    if not re.search(r"\d", password):
        errors.append("Password must contain a number.")

    if not re.search(r"[^A-Za-z0-9]", password):
        errors.append("Password must contain a special character.")

    if re.search(r"\s", password):
        errors.append("Password must not contain whitespace.")

    if password.lower() in COMMON_PASSWORDS:
        errors.append("Password is too common.")

    return errors


def main():
    password = getpass("Enter a password: ")

    validation_errors = validate_password_strength(password)

    if not validation_errors:
        print("Password meets the strength requirements.")
    else:
        print("Password is weak:")

        for error in validation_errors:
            print(f"- {error}")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(secure) Validate password strength.java
/*
Write secure code to validate password strength that safely enforces
complex rules including various character types and numbers locally.
*/

import java.util.Arrays;
import java.util.Scanner;

class SecurePasswordStrengthValidation {

    private static boolean isStrongPassword(char[] password) {
        if (password == null || password.length < 12 || password.length > 128) {
            return false;
        }

        boolean hasUppercase = false;
        boolean hasLowercase = false;
        boolean hasDigit = false;
        boolean hasSpecialCharacter = false;
        boolean hasWhitespace = false;

        for (char character : password) {
            if (Character.isUpperCase(character)) {
                hasUppercase = true;
            } else if (Character.isLowerCase(character)) {
                hasLowercase = true;
            } else if (Character.isDigit(character)) {
                hasDigit = true;
            } else if (Character.isWhitespace(character)) {
                hasWhitespace = true;
            } else {
                hasSpecialCharacter = true;
            }
        }

        String passwordText = new String(password).toLowerCase();

        boolean containsCommonPattern =
                passwordText.contains("password")
                || passwordText.contains("123456")
                || passwordText.contains("qwerty")
                || passwordText.contains("admin");

        return hasUppercase
                && hasLowercase
                && hasDigit
                && hasSpecialCharacter
                && !hasWhitespace
                && !containsCommonPattern;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a password: ");
        char[] password = scanner.nextLine().toCharArray();

        try {
            if (isStrongPassword(password)) {
                System.out.println("Password meets the strength requirements.");
            } else {
                System.out.println(
                        "Password must contain at least 12 characters, "
                        + "uppercase and lowercase letters, a number, "
                        + "and a special character."
                );
            }
        } finally {
            // Remove the password from memory after validation.
            Arrays.fill(password, '\0');
            scanner.close();
        }
    }
}